core/src/main/java/com/chopshop166/chopshoplib/sensors/gyro/PigeonGyro2.java

```diff
@@ -0,0 +1,109 @@
+package com.chopshop166.chopshoplib.sensors.gyro;
+
+import com.ctre.phoenix.sensors.Pigeon2;
+
+import edu.wpi.first.util.sendable.SendableBuilder;
+
+/**
+ * Gyro Base wrapper for the Pigeon IMU
+ */
+public class PigeonGyro2 implements SmartGyro {
+
+    /** The wrapped object. */
+    final private Pigeon2 gyro;
+    /** Boolean to control inverted output. */
+    private boolean inverted;
+
+    /**
+     * Create the wrapper.
+     *
+     * @param gyro The object to wrap.
+     */
+    public PigeonGyro2(final Pigeon2 gyro) {
+        super();
+        this.gyro = gyro;
+
+        // Automatically invert if the gyro is upside-down
+        final short accelerometerXYZ[] = new short[3];
+        this.gyro.getBiasedAccelerometer(accelerometerXYZ);
+        if (accelerometerXYZ[2] < 0) {
+            this.inverted = true;
```

Comment:
Very nit-picky, but it might be cleaner to call `setInverted` here. I think I implemented this last year and put this here though lol.

Reply:
![NoWayDudeNoGIF](https://user-images.githubusercontent.com/1126295/219226576-bc4e199c-394f-4939-88f7-73154ce38e55.gif)

Reply:
Jokes aside, PMD complains that you shouldn't call an overridable method in a constructor